### PR TITLE
Fix RuntimeError: maximum recursion depth exceeded in cmp happened

### PR DESCRIPTION
**Pull request by KazuCocoa**
Fix _RuntimeError: maximum recursion depth exceeded in cmp happened #342_

`self.__class__.__mro__` returns below:

> (<class 'webdriver_test.SubSubWebDriver'>, <class 'webdriver_test.SubWebDriver'>, <class 'appium.webdriver.webdriver.WebDriver'>, <class 'appium.webdriver.extensions.search_context.AppiumSearchContext'>, <class 'appium.webdriver.extensions.action_helpers.ActionHelpers'>, <class 'appium.webdriver.extensions.activities.Activities'>, <class 'appium.webdriver.extensions.applications.Applications'>, <class 'appium.webdriver.extensions.clipboard.Clipboard'>, <class 'appium.webdriver.extensions.context.Context'>, <class 'appium.webdriver.extensions.device_time.DeviceTime'>, <class 'appium.webdriver.extensions.hw_actions.HardwareActions'>, <class 'appium.webdriver.extensions.images_comparison.ImagesComparison'>, <class 'appium.webdriver.extensions.ime.IME'>, <class 'appium.webdriver.extensions.keyboard.Keyboard'>, <class 'appium.webdriver.extensions.location.Location'>, <class 'appium.webdriver.extensions.network.Network'>, <class 'appium.webdriver.extensions.remote_fs.RemoteFS'>, <class 'appium.webdriver.extensions.screen_record.ScreenRecord'>, <class 'selenium.webdriver.remote.webdriver.WebDriver'>, <class 'appium.webdriver.extensions.search_context.AndroidSearchContext'>, <class 'appium.webdriver.extensions.search_context.BaseSearchContext'>, <type 'object'>)

With current implementation, only `<class 'webdriver_test.SubSubWebDriver'>` is skipped.
`maximum recursion depth exceeded ` error happens if `appium.webdriver.webdriver.WebDriver'` is called.

To avoid the issue, I changed to get `<class 'appium.webdriver.extensions.search_context.AppiumSearchContext'>, <class 'appium.webdriver.extensions.action_helpers.ActionHelpers'>, <class 'appium.webdriver.extensions.activities.Activities'>,....` (get only parents of `appium.webdriver.webdriver.WebDriver`)

### CODE REVIEW

File: appium/webdriver/webdriver.py [hunk 1]
```diff
@@ -595,9 +595,27 @@ def battery_info(self):
...
     # pylint: disable=protected-access
 
+    def _get_webdriver_parents(self, mro):
+        """
+        Get Method Resolution Order after `appium.webdriver.webdriver.WebDriver`
+        to avoid __addCommands loop in `appium.webdriver.webdriver.WebDriver`
+
+        :Args:
+            - Tuple of Method Resolution Order
+        :Raises:
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py]: this is not an expected exception, so I wouldn't describe it in the doc

File: appium/webdriver/webdriver.py [hunk 2]
```diff
@@ -595,9 +595,27 @@ def battery_info(self):
...
+            - ValueError: If the MRO has no 'appium.webdriver.webdriver.WebDriver',
+            it raises 'tuple.index(x): x not in tuple' error
+        """
+        m = tuple(map(lambda x: '{}.{}'.format(x.__module__, x.__name__), mro))
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py]: I'd rather use issubclass

File: appium/webdriver/webdriver.py [hunk 3]
```diff
@@ -595,9 +595,27 @@ def battery_info(self):
...
+
+        return mro[(wd_index + 1):]
+
     def _addCommands(self):
-        # call the overridden command binders from all mixin classes
-        for mixin_class in filter(lambda x: x is not self.__class__, self.__class__.__mro__):
+        # call the overridden command binders from all mixin classes except for
+        # appium.webdriver.webdriver.WebDriver and its sub-classes
+        # https://github.com/appium/python-client/issues/342
+        for mixin_class in self._get_webdriver_parents(self.__class__.__mro__):
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py]: `filter(lambda x: not issubclass(x, self.__class__), self.__class__.__mro__)` should be fine

Comment (KazuCocoa) [reply appium/webdriver/webdriver.py]: Yes! I could not find it...thanks so much

**Pull request merged**